File: code_samples/lua/oop_polymorphism.lua
-- Oop: Polymorphism
-- AI/ML Training Sample

Polymorphism = {}
Polymorphism.__ index = Polymorphism

function Polymorphism:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Polymorphism:process(input)
    self.data = input
end

function Polymorphism:getData()
    return self.data
end

function Polymorphism:validate()
    return #self.data > 0
end

-- Example usage
local instance = Polymorphism:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/go/oop_polymorphism.go
package main

import (
    "fmt"
)

// Oop: Polymorphism
// AI/ML Training Sample

type Polymorphism struct {
    Data string
}

func NewPolymorphism() *Polymorphism {
    return &Polymorphism{
        Data: "",
    }
}

func (s *Polymorphism) Process(input string) {
    s.Data = input
}

func (s *Polymorphism) Validate() bool {
    return len(s.Data) > 0
}

func (s *Polymorphism) GetData() string {
    return s.Data
}

func main() {
    instance := NewPolymorphism()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
